.github/workflows/run_tests.yml

```diff
@@ -121,79 +112,87 @@ jobs:
           curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg && \
-          sudo apt-get update && \
-          sudo apt-get -y install iputils-ping docker-ce docker-ce-cli containerd.io redis-tools=6:6.2.5-1rl1~focal1 redis-server=6:6.2.5-1rl1~focal1
-
-      - name: Copy redisai from docker image
-        run: |
+          sudo apt-get -y update && \
+          sudo apt-get -y install iputils-ping docker-ce docker-ce-cli containerd.io redis-tools=6:6.2.5-1rl1~focal1 redis-server=6:6.2.5-1rl1~focal1 &&
           docker create --name redisai --rm redislabs/redisai:${{ matrix.rai_v }}-cpu-bionic && \
-          docker cp redisai:/usr/lib/redis/modules/redisai.so $HOME &&
           sudo mkdir -p /usr/lib/redis/modules/ &&
-          sudo docker cp redisai:/usr/lib/redis/modules/backends/ /usr/lib/redis/modules/
-
-      - name: Setup local redis servers
+          sudo docker cp redisai:/usr/lib/redis/modules/redisai.so /usr/lib/redis/modules &&
+          sudo docker cp redisai:/usr/lib/redis/modules/backends/ /usr/lib/redis/modules/ &&
+          echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/install/lib" >> $GITHUB_ENV &&
+          echo "REDISAI_MODULES=/usr/lib/redis/modules/redisai.so \
+                TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so \
+                TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so" >> $GITHUB_ENV
+
+      - name: Single-shard DB tests
         run: |
-          redis-server --port 7000 --daemonize yes --cluster-enabled yes --cluster-config-file 7000.conf --protected-mode no --save "" --loadmodule $HOME/redisai.so TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so &
-          redis-server --port 7001 --daemonize yes --cluster-enabled yes --cluster-config-file 7001.conf --protected-mode no --save "" --loadmodule $HOME/redisai.so TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so &
-          redis-server --port 7002 --daemonize yes --cluster-enabled yes --cluster-config-file 7002.conf --protected-mode no --save "" --loadmodule $HOME/redisai.so TF /usr/lib/redis/modules/backends/redisai_tensorflow/redisai_tensorflow.so TORCH /usr/lib/redis/modules/backends/redisai_torch/redisai_torch.so
-
-      - name: Overwrite redis cluster env vars
+          SCRATCH_DIR="single_scratch"
+          rm -rf $SCRATCH_DIR; mkdir $SCRATCH_DIR; pushd $SCRATCH_DIR &&
+          redis-server --port 6380 --daemonize yes \
+                       --logfile "single.log" \
+                       --loadmodule $REDISAI_MODULES &&
+          sleep 5; cat single.log &&
```

Comment:
Can this be removed?

Reply:
If not, is the sleep enough for when CI/CD is busy and running slowly?

Reply:
This has been replaced with a more robust check using `redis-cli ping` (which is the Redis-recommended way of checking whether a shard is up)